org.eclipse.scanning.device.ui/src/org/eclipse/scanning/device/ui/points/ValidateResultsView.java

```diff
@@ -0,0 +1,239 @@
+			public void run() {
+				clear();
+			}
+		};
+		man.add(restart);
+
+		final MenuManager menuMan = new MenuManager();
+		menuMan.add(restart);
+		
+		text.setMenu(menuMan.createContextMenu(text));
+	}
+
+	@Override
+	public void dispose() {
+		if (PageUtil.getPage(getSite())!=null) PageUtil.getPage(getSite()).removeSelectionListener(this);
+		super.dispose();
+	}
+	
+	private void clear() {
+		setThreadSafeText(text, new StyledString(""));
+	}
+	
+	@Override
+	public void selectionChanged(IWorkbenchPart part, ISelection selection) {
+		if (selection instanceof IStructuredSelection) {
+			Object ob = ((IStructuredSelection)selection).getFirstElement();
+
+			if (ob instanceof ValidateResults) {
+				ValidateResults validationResults  = (ValidateResults)ob;
+				update(validationResults);
+			}
+
+		}
+	}
+
+	/**
+	 * Method for updating the display with the validation results
+	 * @param results the ValidationResults object
+	 */
+	public void update(ValidateResults results) {
+		if (results == null) return;
+		
+		try {
+
+    		StyledString styledString = new StyledString();
+        	
+			styledString.append("Results received from '");
+			styledString.append(results.getDeviceName(), FontStyler.BOLD);
+			styledString.append("' at ");
+			styledString.append(new Date(System.currentTimeMillis()).toString(), StyledString.COUNTER_STYLER);
+			styledString.append("\n");
+        	
+			styledString.append(appendResultsToStyledString(results));
+
+            setThreadSafeText(text, styledString);
+            return; 
+
+		} catch (Exception ne) {
+			logger.error("Cannot create validation results", ne);
+			if (ne.getMessage()!=null) {
+				setThreadSafeText(text, ne.getMessage());
+			} else {
+				setThreadSafeText(text, ne.toString());
+			}
+		}
+	}
+	
+	/**
+	 * Turn the results into a styled string for display to user, highlighting key values
+	 * @param results The ValidationResults
+	 * @return a StyledString to display on the view
+	 */
+	private StyledString appendResultsToStyledString(ValidateResults results) {
+		StyledString styledString = new StyledString();
+		if (results.getResults() != null) {
+			if (results.getResults() instanceof String) {
+				// Parse out key information from the raw PVStructure string
+				String resultString = (String)results.getResults();
+	
+				styledString.append(resultString);
+				
+				int durationIndex = resultString.indexOf("double duration");
+				if (durationIndex != -1) {
+					int indexOfNewLine = resultString.substring(durationIndex).indexOf("\n");
+					if (indexOfNewLine == -1) {
+						indexOfNewLine = resultString.substring(durationIndex).length() - 1;
+					}
+			    	styledString.setStyle(durationIndex + "double".length(), indexOfNewLine - "double".length(), FontStyler.BOLD);		    	
+				}
+				
+				int axesToMoveIndex = resultString.indexOf("string[] axesToMove");
+				if (axesToMoveIndex != -1) {
+					int indexOfNewLine = resultString.substring(axesToMoveIndex).indexOf("\n");

```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Put single-quotes around '\n' to use the faster "indexOf(char)" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS3027)